dataland-user-service/src/main/kotlin/org/dataland/datalanduserservice/service/PortfolioService.kt

```diff
@@ -0,0 +1,206 @@
+package org.dataland.datalanduserservice.service
+
+import org.dataland.datalandbackendutils.exceptions.InvalidInputApiException
+import org.dataland.datalanduserservice.entity.PortfolioEntity
+import org.dataland.datalanduserservice.exceptions.PortfolioNotFoundApiException
+import org.dataland.datalanduserservice.model.Portfolio
+import org.dataland.datalanduserservice.model.PortfolioResponse
+import org.dataland.datalanduserservice.repository.PortfolioRepository
+import org.dataland.keycloakAdapter.auth.DatalandAuthentication
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.time.Instant
+import java.util.UUID
+
+/**
+ * Service to manage Portfolio-related business logic
+ */
+@Service
+class PortfolioService
+    @Autowired
+    constructor(
+        private val portfolioRepository: PortfolioRepository,
+    ) {
+        private val logger = LoggerFactory.getLogger(PortfolioService::class.java)
+
+        /**
+         * Checks if the portfolio belongs to the user
+         */
+        @Transactional(readOnly = true)
+        fun existsPortfolioForUser(portfolioId: String): Boolean {
+            val userId = DatalandAuthentication.fromContext().userId
+            val correlationId = UUID.randomUUID().toString()
```

Comment:
Die correlation IDs neu zu erzeugen widerspricht dem Sinn der correlation IDs. Die sollten in dem Moment erstellt werden wenn der Prozess im System beginnt und dann immer weitergereicht werden.